src/routes/safe/components/Apps/components/AddAppForm/FormButtons.tsx

```diff
@@ -0,0 +1,51 @@
+import { Button, Divider } from '@gnosis.pm/safe-react-components'
+import React, { ReactElement, useMemo } from 'react'
+import { useFormState } from 'react-final-form'
+import styled from 'styled-components'
+
+import GnoButton from 'src/components/layout/Button'
+import { SafeApp } from 'src/routes/safe/components/Apps/types.d'
+import { isAppManifestValid } from 'src/routes/safe/components/Apps/utils'
+
+const StyledDivider = styled(Divider)`
+  margin: 16px -24px;
+`
+
+const ButtonsContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+`
+
+interface Props {
+  appInfo: SafeApp
+  onCancel: () => void
+}
+
+const FormButtons = ({ appInfo, onCancel }: Props): ReactElement => {
+  const { valid, validating, visited } = useFormState({
+    subscription: { valid: true, validating: true, visited: true },
+  })
+
+  const isSubmitDisabled = useMemo(() => {
+    // if non visited, fields were not evaluated yet. Then, the default value is considered invalid
+    const fieldsVisited = visited?.agreementAccepted && visited.appUrl
```

Comment:
I know that if first element doesn't meet the condition because `visited` is undefined won't pass the && but I would add the ? in the second visited, just in case we remove the first condition